### PR TITLE
Add tool to upload a block to local development environment

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR adds a script that will upload a block to a local development environment for testing and debugging.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated.
- [n/a] Documentation added.
- [n/a] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [n/a] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by flxbk — APPROVED** [review]

File: development/tools/upload-block/upload-block.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  cat <<EOF | aws_with_creds s3 cp - "s3://mimir-tsdb/$TENANT/markers/$BLOCK_ULID-no-compact.json"
+  {
+    "id": "$BLOCK_ULID",
+    "version": 1,
+    "no_compact_time": $(date -u +%s),
+    "reason": "manual",
+    "details": "block uploaded for debugging purposes"
+  }
+EOF
```
Comment (pstibrany) [inline development/tools/upload-block/upload-block.sh:33]: we have `markblocks` tool that could be used, if it skipped the blocks-exists check.

For correctness, this marker file should also be in the block directory itself.

File: development/tools/upload-block/upload-block.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  fi
+
+  local BLOCK_DIR="$1"
+
+  if [ ! -d "$BLOCK_DIR" ]; then
+    echo "'$BLOCK_DIR' is not a directory." >/dev/stderr
+    exit 1
+  fi
+
+  BLOCK_ULID="$(basename "$BLOCK_DIR")"
```
Comment (pstibrany) [inline development/tools/upload-block/upload-block.sh:20]: If `BLOCK_ULID` isn't valid ulid, uploaded block will be ignored. We could check for it by calling `ulidtime` (from mimir/tools/ulidtime) and warn if it fails.

**Review by pstibrany — APPROVED** [review]
Left some nit comments. Thank you.

File: development/tools/upload-block/upload-block.sh (hunk 1 above)
Comment (charleskorn) [reply development/tools/upload-block/upload-block.sh]: Good point, I've switched to using `markblocks` in 28f3244.

File: development/tools/upload-block/upload-block.sh (hunk 2 above)
Comment (charleskorn) [reply development/tools/upload-block/upload-block.sh:26]: I've added a check in ff0d022.

**Pull request merged**